src/pseudopeople/configuration/generator.py

```diff
@@ -75,12 +75,15 @@ def get_configuration(user_configuration: Union[Path, str, Dict] = None) -> Conf
     :param user_configuration: A path to the YAML file or a dictionary defining user overrides for the defaults
     :return: a ConfigTree object of the noising configuration
     """
-
-    if type(user_configuration) == str and user_configuration.lower() == Keys.NO_NOISE:
-        config_type = Keys.NO_NOISE
+    if isinstance(user_configuration, str) and user_configuration.lower() == NO_NOISE:
+        config_type = NO_NOISE
         user_configuration = None
+    elif isinstance(user_configuration, (Path, str)):
+        with open(user_configuration, "r") as f:
+            user_configuration = yaml.full_load(f)
+        config_type = DEFAULT
     else:
-        config_type = Keys.DEFAULT
+        config_type = DEFAULT
```

Comment:
I'm fairly confident there is a cleaner version of this if/elif/else block but couldn't get one to work.